src/Sarif.Converters/BuiltInConverterFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    // Factory class for creating one of the built-in converters.
+    internal class BuiltInConverterFactory : ConverterFactory
+    {
+        private static readonly IDictionary<string, Lazy<ToolFileConverterBase>> BuiltInConverters = CreateBuiltInConverters();
+
+        public override ToolFileConverterBase CreateConverterCore(string toolFormat)
+        {
+            Lazy<ToolFileConverterBase> converter;
+            return BuiltInConverters.TryGetValue(toolFormat, out converter)
+                ? converter.Value
+                : null;
+        }
+
+        private static Dictionary<string, Lazy<ToolFileConverterBase>> CreateBuiltInConverters()
+        {
+            var result = new Dictionary<string, Lazy<ToolFileConverterBase>>();
+            // TODO: Reflect to get them all.
```

Comment:
>// TODO: Reflect to get them all. [](start = 12, length = 33)

Disagree. Breaks my first principle for for use reflection, if you can get away with not using it, don't use it. What I'd suggest instead is that we either create a #if DEBUG block that asserts we understand and have hard-coded for all existing built-in converters, or that we alter a unit test that fires if reflection doesn't match what is hard-coded. If you weren't using generics, it'd be easy to create an array of ToolFormat values below that unit-tests could compare with reflected values. I think you'd have to rework your API to accomplish this. Most straightforward way would be to have an api that understands ToolFormat and which has a switch statement, each case of which is away of the type of the converter to create. This might actually be nice in any case, since the way you've done it allows for decoupling between the converter type and the toolformat value...